netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -206,6 +192,40 @@ public void deframeFailed(Throwable cause) {
       handler.getWriteQueue().enqueue(new CancelServerStreamCommand(this, status), true);
     }
 
+    private void onWriteFrameData(ChannelFuture future, int numMessages, int numBytes) {
+      // Remove the bytes from outbound flow control, optionally notifying
+      // the client that they can send more bytes.
+      // TODO(sergiitk): should onSentBytes be called only on success?
```

Comment:
@ejona86 this is might be out-of-scope of this PR, but I've noticed that the client side only calls it when the future has succeeded: https://github.com/grpc/grpc-java/blob/f7ee5f3182dddde775069155d17ed93f7270f1ca/netty/src/main/java/io/grpc/netty/NettyClientStream.java#L191-L196

While server-side does not:
https://github.com/grpc/grpc-java/blob/f7ee5f3182dddde775069155d17ed93f7270f1ca/netty/src/main/java/io/grpc/netty/NettyServerStream.java#L116-L124

I assume this was based on the assumption the promises won't fail. I think we should move `onSentBytes` into `if (future.isSuccess())`.